sdk/python/kfp/compiler/compiler.py

```diff
@@ -223,9 +223,12 @@ def _process_args(self, raw_args, argument_inputs):
       matches += _match_serialized_pipelineparam(str(processed_args[i]))
       unsanitized_argument_inputs = {}
       for x in list(set(matches)):
-        sanitized_str = str(dsl.PipelineParam(K8sHelper.sanitize_k8s_name(x[1]), K8sHelper.sanitize_k8s_name(x[0]), x[2]))
-        unsanitized_argument_inputs[sanitized_str] = str(dsl.PipelineParam(x[1], x[0], x[2]))
-
+        if len(x) == 3 or (len(x) == 4 and x[3] == ''):
```

Comment:
nit: could you add some comments on what's the expected data and length on the x? It's hard to understand what the code does from hardcoded indexes.

Reply:
Done.